Modules/_ctypes/_ctypes.c

```diff
@@ -3094,6 +3116,9 @@ static PyType_Spec pycdata_spec = {
 static int
 PyCData_MallocBuffer(CDataObject *obj, StgInfo *info)
 {
+    /* We don't have to lock in this function, because it's only
+       used in constructors and therefore does not have concurrent
+       access. */
```

Comment:
Let's fail if this changes.
```suggestion
    /* We don't have to lock in this function, because it's only
     * used in constructors and therefore does not have concurrent
     * access.
     */
   assert (Py_REFCNT(obj) == 1);

```
(This might fail on a benign change, but we can deal with that when it does.)